# HG changeset patch
# User alta88@fixall.com
# Date 1740081430 25200
# Parent  d2f8604747673af16c449f85ac6f3a5878922c77
Feed enhancements
- For Atom feeds, like for RSS already, look for enclosures in <media> tags.

diff --git a/mailnews/extensions/newsblog/FeedParser.sys.mjs b/mailnews/extensions/newsblog/FeedParser.sys.mjs
--- a/mailnews/extensions/newsblog/FeedParser.sys.mjs
+++ b/mailnews/extensions/newsblog/FeedParser.sys.mjs
@@ -1008,9 +1008,10 @@ FeedParser.prototype = {
       }
 
       item.xmlContentBase = this.validLink(item.xmlContentBase);
 
-      // Handle <link rel="enclosure"> (if present).
+      // Handle <link rel="enclosure"> (if present) and <media:content>, which
+      // may be in a <media:group> (if present).
       tags = this.childrenByTagNameNS(
         itemNode,
         lazy.FeedUtils.ATOM_IETF_NS,
         "link"
@@ -1038,8 +1039,23 @@ FeedParser.prototype = {
           }
         }
       }
 
+      tags = itemNode.getElementsByTagNameNS(lazy.FeedUtils.MRSS_NS, "content");
+      if (tags) {
+        for (const tag of tags) {
+          const url = this.validLink(tag.getAttribute("url"));
+          if (url && !encUrls.includes(url)) {
+            const type = this.removeUnprintableASCII(tag.getAttribute("type"));
+            const fileSize = this.removeUnprintableASCII(
+              tag.getAttribute("fileSize")
+            );
+            item.enclosures.push(new lazy.FeedEnclosure(url, type, fileSize));
+          }
+        }
+      }
+
+      // The <origEnclosureLink> tag - see above.
       tags = this.childrenByTagNameNS(
         itemNode,
         lazy.FeedUtils.FEEDBURNER_NS,
         "origEnclosureLink"
